### ISSUE TITLE
Skipped tests not shown in jUnit output when status-level="all"

### ISSUE THREAD

**Issue by detly** [nextest-rs/nextest#885]
The jUnit XML report does not seem to include skipped tests, even when the `status-level` is set to `"all"` in `.config/nextest.toml`.

Using the default project generated by `cargo init --lib`, I have this in `src/lib.rs`:

```rust
pub fn add(left: usize, right: usize) -> usize {
    left + right
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    #[ignore]
    fn it_works() {
        let result = add(2, 2);
        assert_eq!(result, 4);
    }
}
```

...and this in `.config/nextest.toml`:

```toml
[profile.default]
status-level = "all"
final-status-level = "all"
fail-fast = false

[profile.default.junit]
path = "test-results.xml"
```

The generated XML is in `target/nextest/default/test-results.xml` as expected, but contains:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="nextest-run" tests="0" failures="0" errors="0" uuid="83992729-11dc-43e7-b5e6-58baf30be039" timestamp="2023-06-08T02:13:52.673+00:00" time="0.000">
</testsuites>
```

Skipped tests **are** reported on `stdout` though:

```none
nextest-test ⚬ cargo nextest run
    Finished test [unoptimized + debuginfo] target(s) in 0.00s
    Starting 0 tests across 1 binaries (1 skipped)
        SKIP [         ] nextest-test tests::it_works
------------
     Summary [   0.000s] 0 tests run: 0 passed, 1 skipped
        SKIP [         ] nextest-test tests::it_works
```

I had expected skipped tests to also be listed in the jUnit output.

OS: Ubuntu 22.10
Cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
Rustc: 1.70.0 (90c541806 2023-05-31)
Nextest: 0.9.53 (installed via `cargo install` for dev, pre-built binary for CI)

**Comment by sunshowers** [comment]
Hi there! Yes, we should have a reasonable depiction of skipped tests in the JUnit output.

Would you like to work on this? I'd love to work out a design with you—if you have a general suggestion, we can take it from there. In particular, should we expose this as a config option or just include skipped tests unconditionally?

**Comment by detly** [comment]
Oof, I'm sorry, I did not see a notification for this! I am happy to chip away at it, and yes, let's start with at least a vague design.

> should we expose this as a config option or just include skipped tests unconditionally

First off - do you have any particular policy (or just expectations) around backwards compatibility or unexpected output changes? Because that would more or less decide it ie. it'd need to be a default-to-current-behaviour config option. Otherwise I'd go with making it include the skipped tests unconditionally, to make it consistent with the stdout reporting and the docs.

**Comment by sunshowers** [comment]
I think we do generally want machine-readable output to be stable, modulo bugs. This seems like a case where it would be pretty surprising to users to suddenly start seeing lots of extra tests in their dashboards, so I think putting it behind a config option makes sense.

**Comment by sirynka** [comment]
Seems like this bit is responsible for skipped tests. If we'd uncomment it and update the naming the feature would work.
But I have not checked how to hide it behind the config flag yet.

https://github.com/nextest-rs/nextest/blob/6304b41ee0d07c2fdcb9d7b89d04b9c934eac0b2/nextest-runner/src/reporter/aggregator.rs#L192-L203

**Comment by sunshowers** [comment]
Yes, that looks right. So it seems like it should be easy to do, we just need to figure out the configuration UI surface for this.

**Comment by rursprung** [comment]
i just stumbled across this. i understand the reasoning for intending to make it optional, but once it's implemented the default should probably be switched with the next major release after that.

(i'm aware that this isn't implemented yet; just adding this as a "IMHO that should be done after implementing it")
